LearningJourney/Modules/Features/JLibrary/JLibraryTests/Sources/Scenes/Library/Domain/UseCases/SignoutUseCaseTests.swift

```diff
@@ -0,0 +1,70 @@
+import XCTest
+@testable import JLibrary
+import CoreAdapters
+import CoreAuthentication
+
+final class SignoutUseCaseTests: XCTestCase {
+    // MARK: - Properties
+
+    private let tokenCacheSpy = TokenCacheSpy()
+    private let notificationCenterSpy = NotificationCenterSpy()
+    private lazy var sut = SignoutUseCase(cache: tokenCacheSpy, notificationCenter: notificationCenterSpy)
+
+    // MARK: - Unit tests
+
+    func test_execute_itShouldNotifyAuthChange() {
+        // Given
+
+        // When
+
+        sut.execute()
+
+        // Then
+
+        XCTAssertEqual(notificationCenterSpy.postedNotificationsCount, 1)
+    }
+
+    func test_execute_itShouldClearCache() {
+        // Given
+
+        // When
+
+        sut.execute()
+
+        // Then
+
+        XCTAssertEqual(tokenCacheSpy.cacheClearedCount, 1)
+    }
```

Comment:
This should be a single test
```suggestion
    func test_execute_itShouldNotifyAuthChange_andClearToken() {
        // Given / When

        sut.execute()

        // Then
        XCTAssertEqual(notificationCenterSpy.postedNotificationsCount, 1)
        XCTAssertEqual(tokenCacheSpy.cacheClearedCount, 1)
    }
```